cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DistributedStorageProviderNamespaceAdmin.java

```diff
@@ -55,15 +65,40 @@ public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreExcep
     super.create(namespaceMeta);
     // TODO: CDAP-1519: Create base directory for filesets under namespace home
     // create HBase namespace
-    tableUtil.createNamespaceIfNotExists(getAdmin(), namespaceMeta.getNamespaceId().toId());
```

Comment:
just a thought: with HDFS and Hive stuff also being added in these methods, do you think the code will be more readable if each component has its own dedicated private methods for create and delete? I'm not sure we need separate classes for handling these right now, but methods?